src/pj_factors.c

```diff
@@ -17,12 +17,18 @@ int pj_factors(LP lp, const PJ *P, double h, struct FACTORS *fac) {
     PJ_COORD coo = {{0, 0, 0, 0}};
     coo.lp = lp;
 
-    err = proj_errno_reset (P);
+    /* These are probably due to earlier errors, so we leave errno alone */
+    if (0==fac)
```

Comment:
But this is C89, not C++, and a pretty common idiom. I have been using it for "clarity through brevity" for many years. It was actually my understanding that **by stdc definition** a `0` interpreted as a pointer  would have to expand to the null pointer, even though the actual hardware representation of that might be different from integer `0`. 

But `NULL` is a macro that expands to `(void *) 0`, and I have recently become in doubt whether it is actually strictly true that a plain `0`, interpreted as a pointer, is identical to a plain `0` cast as `void *`.  I will have to do some standards studying.

But if strictly `NULL` (i.e. `(void *) 0`) is needed here, it will break a common idiom (and hence a lot of code).

Nevertheless, it is valuable to know that this idiom is confusing, so I will retract to my monastery to contemplate, and may return enlightened...

Reply:
Ok, that answers my question about which version of C.  The C++ community pushed for 0 over NULL or a long time and then when running for the typed nullptr.   nullptr helped catch quite a few long term lurking bugs.

I suggesting NULL to make it clear that the return is a pointer and not a boolean false or something else.  I think that if the 0 is applied to a pointer value, it should always be safely  converted to a `NULL`.  But I think you are correct that 0 assigned to a ptr always is the same as NULL.

Some quickly found related readings:

C:
- http://en.cppreference.com/w/c/types/NULL
- https://stackoverflow.com/questions/1296843/what-is-the-difference-between-null-0-and-0

C++:
- https://stackoverflow.com/questions/36484473/is-c-null-equal-to-c11-nullptr
- https://stackoverflow.com/questions/20509734/null-vs-nullptr-why-was-it-replaced